src/lib/replication/service/raft_repl_service.cpp

```diff
@@ -247,6 +255,40 @@ AsyncReplResult< shared< ReplDev > > RaftReplService::create_repl_dev(group_id_t
                   : make_async_error< shared< ReplDev > >(ReplServiceError::SERVER_NOT_FOUND);
 }
 
+// Remove repl_dev for raft is a 3 step process.
+// Step 1: The leader of the repl_dev proposes all members an entry to delete the entire group. As part of commit of
+// this entry, all members call leave() method, which does the following:
+//      a) Mark in cache that the group is destroyed, so no subsequent write operations are permitted (relevant to
+//      leader only)
+//      b) Call the upper_layer on_destroy(), so that consumers persistent meta information can be deleted.
```

Comment:
it requires consumer to do its on_destroy in a sync way, right?

Reply:
Consumer can do on_destroy in async way as long as they persist that information in their metablk.

Reply:
as long as they persist that information in their metablk "before returning on_destroy()", correct? 
We better make return type as future and homestore shouldn't proceed without a promise is made. Otherwise, there will be resource leak. If we leave it as return void, it has to to written in API contract that all destroy has to be completed before returning to homestore.